s2fft/utils/jax_pritimive.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+from functools import partial
+from typing import Tuple
+
+import jax.numpy as jnp
+from jax import jit, lax
+from jax._src.api import ShapeDtypeStruct
+from jax._src.core import ShapedArray
+from jax._src.typing import Array, ArrayLike
```

Comment:
We should be primarily  importing from JAX's [public API ](https://jax.readthedocs.io/en/latest/api_compatibility.html) not from the `jax._src` subpackage which is not considered part of the public API and can be changed without warning. While we may need to rely on some modules not in the public API, these types all seem to be available in the public API under different modules so should be imported from there.